veps/sig-network/decouple-nad.md

```diff
@@ -0,0 +1,183 @@
+
+Currently, virt-controller queries the NAD CR for every secondary network in every VM. It does so by querying the API server each time it templates a new virt-launcher pod, which occurs during initial VM startup and every migration.
+`NetworkAttachmentDefinition` is an external CRD maintained by the [k8s network plumbing WG](https://github.com/k8snetworkplumbingwg). The mentioned query is the only direct dependency that KubeVirt production code has on this external project.
+This proposal aims to remove that query, in favor of an external replacement that will map the custom resources to the virt-launcher pod instead.  
+In addition, the virt-controller and virt-operator role bindings that currently allow it to access the NAD can be reduced in scope.
+
+
+## Motivation
+
+The replacement has the architectural advantage of decoupling KubeVirt from an external dependency by simplifying code maintenance and supporting future replacement technologies.
+Security will be improved by removing access permissions to the external CRD.
+In addition, a performance improvement can be gained from using a project that caches NAD data. 
+    
+
+## Goals
+
+- KubeVirt will no longer be aware of NetworkAttachmentDefinition objects.
+- Decrease the number of API calls required to spin-up and migrate a VM
+
+## Non Goals
+
+Removal of other multus related dependencies 
+
+## Definition of Users
+
+This enhancement is intended for:
+- VM owners
+- Namespace owners
+- Cluster Admins
+
+## User Stories
+
+- As a VM owner, I want to accelerate VM load time
+- As a VM owner and a namespace owner, I want my system to use the principle of least privilege, such that service accounts will only have access to resources that they absolutely need. 
+- As a Cluster Admin I want to reduce the use of ClusterRoles/Bindings as much as possible, so that my system would use the principle of least privilege.
+
+## Repos
+
+https://github.com/kubevirt/kubevirt
+
+## Design
+
+### Current Architecture
+KubeVirt currently has a tight coupling with NetworkAttachmentDefinition (NAD) resources. When a VM with secondary networks is launched, the virt-controller directly queries NAD CRs to extract custom resource requirements ( `k8s.v1.cni.cncf.io/resourceName` annotation) and injects them into the virt-launcher pod specification, as resource requests and limits.
+
+### Proposed Architecture
+The new architecture decouples KubeVirt from NAD resources by leveraging the network-resources-injector as an external admission controller. This creates a cleaner separation of concerns where KubeVirt focuses on VM lifecycle management while network-resources-injector handles network resource injection.
```

Comment:
Since `network-resources-injector` is first introduced here, consider adding a link to its GitHub repository and providing a brief explanation of what it does.